src/frameworks/next-intl.ts

```diff
@@ -0,0 +1,106 @@
+import { TextDocument } from 'vscode'
+import { Framework, ScopeRange } from './base'
+import { LanguageId } from '~/utils'
+import { RewriteKeySource, RewriteKeyContext, KeyStyle } from '~/core'
+
+class NextIntlFramework extends Framework {
+  id = 'next-intl'
+  display = 'next-intl'
+  namespaceDelimiter = '.'
+  perferredKeystyle?: KeyStyle = 'nested'
+
+  namespaceDelimiters = ['.']
+  namespaceDelimitersRegex = /[\.]/g
+
+  detection = {
+    packageJSON: [
+      'next-intl',
+    ],
+  }
+
+  languageIds: LanguageId[] = [
+    'javascript',
+    'typescript',
+    'javascriptreact',
+    'typescriptreact',
+    'ejs',
+  ]
+
+  usageMatchRegex = [
+    // Basic usage
+    '[^\\w\\d]t\\([\'"`]({key})[\'"`]',
+
+    // Rich text
+    '[^\\w\\d]t\.rich\\([\'"`]({key})[\'"`]',
+
+    // Raw text
+    '[^\\w\\d]t\.raw\\([\'"`]({key})[\'"`]',
+  ]
+
+  refactorTemplates(keypath: string) {
+    // Provide options where the last segment is the keypath.
+    // Ideally we'd automatically consider the namespace here.
+    const lastSegment = keypath.split('.').pop()
+
+    return [
+      `{t('${lastSegment}')}`,
+      `{t('${keypath}')}`,
+      `t('${lastSegment}')`,
+      `t('${keypath}')`,
+    ]
```

Comment:
I've added the `lastSegment` options because typically you use `next-intl` with a namespace.

Is it possible to consider an existing namespace in this function, to automatically remove the namespace from the suggestion?

Also if the component uses a given namespace, it would be great if that would show up as a default when entering a key. Is this possible?

Reply:
Namespaces were hard to implement with the current architecture. Let's get the framework running as is now